housekeeper/store/filters/file_join_filters.py

```diff
@@ -0,0 +1,41 @@
+from enum import Enum
+from typing import Callable, List, Optional
+from sqlalchemy.orm import Query
+from sqlalchemy import func as sqlalchemy_func
+
+from housekeeper.store.models import File, Tag
+
+
+def filter_files_by_tags(files_query: Query, tag_names: List[str], **kwargs) -> Query:
+    """Filter files by tag names."""
+    return (
+        files_query.filter(Tag.name.in_(tag_names))
+        .group_by(File.id)
+        .having(sqlalchemy_func.count(Tag.name) == len(tag_names))
+    )
+
+
+def filter_files_by_archive(files_query: Query, is_archived: bool, **kwargs) -> Query:
+    """Filters the query depending on if the files are archived or not."""
+    return files_query.filter((File.archive != None) == is_archived)
```

Comment:
This logic is difficult to grasp, can this be rewritten to make it clearer?

Reply:
I couldn't make the intuitive python logic work unfortunately. With good tests and docstring I think this is okay.